src/AppBundle/Command/AddUserCommand.php

```diff
@@ -0,0 +1,265 @@
+ *
+ * @author Javier Eguiluz <javier.eguiluz@gmail.com>
+ */
+class AddUserCommand extends ContainerAwareCommand
+{
+    const MAX_ATTEMPTS = 5;
+
+    /** @var ObjectManager */
+    private $em;
+
+    protected function configure()
+    {
+        $this
+            // a good practice is to use the 'app:' prefix to group all your custom application commands
+            ->setName('app:add-user')
+            ->setDescription('Creates users and stores them in the database')
+            ->setHelp($this->getCommandHelp())
+            // commands can optionally define arguments and/or options (mandatory and optional)
+            // see http://symfony.com/doc/current/components/console/console_arguments.html
+            ->addArgument('username', InputArgument::OPTIONAL, 'The username of the new user')
+            ->addArgument('password', InputArgument::OPTIONAL, 'The plain password of the new user')
+            ->addArgument('email', InputArgument::OPTIONAL, 'The email of the new user')
+            ->addOption('is-admin', null, InputOption::VALUE_NONE, 'If set, the user is created as an administrator')
+        ;
+    }
+
+    /**
+     * This method is executed before initialize() and execute(). Its purpose is
+     * to check if some of the options/arguments are missing and interactively
+     * ask the user for those values.
+     *
+     * This method is completely optional. If you are developing an internal console
+     * command, you probably should not implement this method because it requires
+     * quite a lot of work. However, if the command is meant to be used by external
+     * users, this method is a nice way to fall back and prevent errors.
+     */
+    protected function interact(InputInterface $input, OutputInterface $output)
+    {
+        if (null !== $input->getArgument('username') && null !== $input->getArgument('password') && null !== $input->getArgument('email')) {
+            return;
+        }
+
+        $output->writeln('');
+        $output->writeln('Add User Command Interactive Wizard');
+        $output->writeln('-----------------------------------');
+
+        $output->writeln(array(
+            '',
+            'If you prefer to not use this interactive wizard, provide the',
+            'arguments required by this command as follows:',
+            '',
+            ' $ php app/console app:add-user username password email@example.com',
+            '',
+        ));
+
+        $output->writeln(array(
+            '',
+            'Now we\'ll ask you for the value of all the missing command arguments.',
+            '',
+        ));
+
+        // See http://symfony.com/doc/current/components/console/helpers/questionhelper.html
+        $console = $this->getHelper('question');
+
+        // Ask for the username if it's not defined
+        if (null === $username = $input->getArgument('username')) {
+            $question = new Question(' > <info>Username</>: ');
```

Comment:
why not use `</info>`? It's much better to be explicit imo (and you know the start tag, as it's in the same string)